src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MemberAnalysisResult.cs

```diff
@@ -24,25 +28,22 @@ internal struct MemberAnalysisResult
         /// </summary>
         public readonly bool HasAnyRefOmittedArgument;
 
-        private MemberAnalysisResult(MemberResolutionKind kind)
-            : this(kind, default(ImmutableArray<int>), default(ImmutableArray<int>), default(ImmutableArray<Conversion>))
-        {
-        }
-
         private MemberAnalysisResult(
             MemberResolutionKind kind,
-            ImmutableArray<int> badArgumentsOpt,
-            ImmutableArray<int> argsToParamsOpt,
-            ImmutableArray<Conversion> conversionsOpt,
+            ImmutableArray<int> badArgumentsOpt = default,
+            ImmutableArray<int> argsToParamsOpt = default,
+            ImmutableArray<Conversion> conversionsOpt = default,
             int missingParameter = -1,
-            bool hasAnyRefOmittedArgument = false)
+            bool hasAnyRefOmittedArgument = false,
+            ImmutableArray<TypeParameterDiagnosticInfo> constraintFailureDiagnosticsOpt = default)
         {
             this.Kind = kind;
             this.BadArgumentsOpt = badArgumentsOpt;
             this.ArgsToParamsOpt = argsToParamsOpt;
             this.ConversionsOpt = conversionsOpt;
             this.BadParameter = missingParameter;
             this.HasAnyRefOmittedArgument = hasAnyRefOmittedArgument;
+            this.ConstraintFailureDiagnosticsOpt = constraintFailureDiagnosticsOpt.NullToEmpty();
```

Comment:
>ConstraintFailureDiagnosticsOpt [](start = 17, length = 31)

`ConstraintFailureDiagnosticsOpt` shouldn't be "Opt" if we always at least put an empty array there.